test/environment_test/test.cpp

```diff
@@ -24,104 +24,108 @@ static_assert(gnc::constant::init_gps_time_of_week == 0,
 static_assert(gnc::constant::init_gps_nanoseconds == 0,
     "Tests are only applicable if the GPS nanoseconds is 0!");
 
-#define ASSERT_VEC_NEAR(tol, a, b) ASSERT_NEAR(lin::fro(a - b), 0.0, tol)
-
-TEST(EnvironmentTest, EarthAttitudeTest) {
+void test_env_earth_attitude() {
   lin::Vector3d w;
   lin::Vector4d q;
   // Calculate and check for t=0.0 (comparing against MATLAB)
   gnc::env::earth_attitude(0.0, q);
   gnc::env::earth_angular_rate(0.0, w);
-  ASSERT_VEC_NEAR(1e-7, q, lin::Vector4d({
+  TEST_ASSERT_DOUBLE_VEC_NEAR(1e-7, q, lin::Vector4d({
```

Comment:
Now that I think about it--I'm wondering if there may be numerical issues that cause lin to fail on Teensy? Is that possible?

Reply:
The tests failing don't even use lin - currently only `ode23` is failing. It has to come down to a compiler difference between clang and GNU. I've had my friend build on different laptops and it only fails with GNU compilers - really beats me why.